src/safeds/data/tabular/transformation/_discretizer.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from sklearn.preprocessing import KBinsDiscretizer as sk_KBinsDiscretizer
+
+from safeds.data.tabular.containers import Table
+from safeds.data.tabular.transformation._table_transformer import TableTransformer
+from safeds.exceptions import TransformerNotFittedError, UnknownColumnNameError
+
+
+class Discretizer(TableTransformer):
+    """
+    The Discretizer transforms replaces a continuous variable by a variable that only has a finite amount of values.
+
+    Parameters
+    ----------
+    number_of_bins: float
+        The number of bins to be created.
+
+    Raises
+    ------
+    ValueError
+        If the given number_of_bins is less than 2.
+    """
+
+    def __init__(self, number_of_bins: float = 5):
+        self._column_names: list[str] | None = None
+        self._wrapped_transformer: sk_KBinsDiscretizer | None = None
+
+        if number_of_bins < 2:
+            raise ValueError("Parameter 'number_of_bins' must be >= 2.")
+        self.number_of_bins = number_of_bins
+
+    def fit(self, table: Table, column_names: list[str] | None) -> Discretizer:
+        """
+        Learn a transformation for a set of columns in a table.
+
+        This transformer is not modified.
+
+        Parameters
+        ----------
+        table : Table
+            The table used to fit the transformer.
+        column_names : Optional[list[str]]
+            The list of columns from the table used to fit the transformer. If `None`, all columns are used.
+
+        Returns
+        -------
+        fitted_transformer : TableTransformer
+            The fitted transformer.
+        """
+        if column_names is None:
+            column_names = table.column_names
+        else:
+            missing_columns = set(column_names) - set(table.column_names)
+            if len(missing_columns) > 0:
+                raise UnknownColumnNameError(list(missing_columns))
+
+        wrapped_transformer = sk_KBinsDiscretizer(n_bins=self.number_of_bins)
```

Comment:
```suggestion
        wrapped_transformer = sk_KBinsDiscretizer(n_bins=self._number_of_bins, encode='ordinal')
```
Use the ordinal encoding. This transforms the data and returns the bin index rather than a bin as sparse matrix. This will also resolve the problem with the tests.